tests/compile_asserts.py

```diff
@@ -24,43 +25,57 @@ def compile_and_save(approval, version, test_name):
     return tealdir, name, compiled
 
 
-def mismatch_ligature(expected, actual):
-    la, le = len(actual), len(expected)
-    mm_idx = -1
-    for i in range(min(la, le)):
-        if expected[i] != actual[i]:
-            mm_idx = i
-            break
-    if mm_idx < 0:
-        return ""
-    return " " * (mm_idx) + "X" + "-" * (max(la, le) - mm_idx - 1)
+def process_line(line: str) -> str:
+    quote_types = ("'", '"')
+    comment_start = "/"
 
+    processed_end = len(line) - 1
 
-def assert_teal_as_expected(path2actual, path2expected):
+    current_quote: str | None = None
+    for i, char in enumerate(line):
+        for qt in quote_types:
+            if char == qt:
```

Comment:
Actually this probably needs to take into account escaped quotes, e.g. `\"`

Reply:
I'm not sure the purpose of processing each line, versus just keeping the raw original. IMO, for this test code it's okay (even preferrable?) to fail because the generated teal is non-identical, even if it's semantically equivalent. I think we would want to know when this happens, and in that case we could decide to simply replace the old fixtures with the new ones.

Reply:
Ok, I remember now. When I originally wrote this test, I had some hand-commented code which I didn't want to fail the diff. I think at this point, the usage has grown much larger than I originally anticipated. So the comments I just wrote, convince me that it's better to _not_ try and pre-process. I.e., don't ignore comments. If we agree that this approach is reasonable, I'm more than happy to replace the original fixtures with fixtures which will pass the tests.

Reply:
I agree that it's preferable to fail if the generated code is different (otherwise we shouldn't be checking in any code for this).

Some checked in code is augmented with comments, so a direct line by line comparison will fail, even though the ops are the same. This processing function is responsible for stripping away the comments plus any difference in whitespace.

I haven't looked in detail at all the code, but my impression is we'd want to keep the additional user-generated comments if at all possible.

Reply:
Ah, looks like we replied at the same time. If you're ok with getting rid of the comments, I'm happy to make that change to simplify this comparison code

Reply:
Yeah for sure. Those original old fixtures don't need their comments anymore.

Reply:
Updated in 27f530e